projects/browser-extension/vue-app/src/store/index.js

```diff
@@ -74,6 +78,30 @@ export default createStore({
 
       commit('SET_SITEMAPS', { sitemaps });
     },
+    async GET_OPENSEARCH_URL ({ commit, state }) {
+      const href = findLinkHref(state.head, 'search');
+
+      if (href) {
+        const url = createAbsoluteUrl(state.head, href);
+        commit('SET_OPENSEARCH_URL', { url });
+      } else {
+        commit('SET_OPENSEARCH_URL', { url: null });
+      }
+    },
+    async GET_OPENSEARCH_FILE ({ commit, dispatch, state }) {
+      await dispatch('GET_OPENSEARCH_URL');
```

Comment:
You could consider returning something from the `GET_OPENSEARCH_URL` action. Just a different pattern to dispatch and get the url into the scope of the `GET_OPENSEARCH_FILE ` action.